proposals/wg-image-compatibility.md

```diff
@@ -0,0 +1,106 @@
+These are usually described in the release notes or installation guides. We believe we have identified the missing puzzle in the OCI standard,
+the image compatibility specification, which is a key feature that describes the requirements of special containerized applications
+that have hard requirements for specific kernel versions, configurations, modules, or out-of-tree drivers.
+
+For example, a container application  free5GC UPF requiring the gtp5g kernel module  will only be compatible with kernel 5.0.0-23-generic 
+or 5.4.x due to the module's hard kernel version requirements - https://free5gc.org/guide/3-install-free5gc/.
+If SR-IOV or GPU is required by the container, other requirements have to be specified against the host OS.
+Kata Containers use cases are the best example: https://github.com/kata-containers/kata-containers/tree/main/docs/use-cases.
+
+
+The described incompatibility problem affects distributed systems.
+In multi-cloud, hybrid cloud, or even private cloud scenarios, one containerized cloud-native application will need to run
+on different Kubernetes distributions that use different host operating systems on different worker nodes.
+
+Operating systems can use different kernel versions with different configurations.
+Different operating system distributions may integrate different devices and their corresponding drivers,
+even for the same device different versions of drivers may be used. Therefore, different operating system distributions are not identical.
+Even within the same OS distro, different releases are also not identical.
+
+An image compatibility specification would help container image authors describe compatibility requirements in a standard way.
+The specification will be uploaded with the image to the image registry. This makes hard container compatibility requirements discoverable,
+programmable, and will support different consumers and cover use cases where the application requires a specific compatible environment.
+
+For further reference please follow:
+* [OCI Image Compatibility - the spec reasoning](https://docs.google.com/presentation/d/1F9GnCm2sULuyTJ5BEFZlL8Qjab81DK7g9Oy6qBfe5Qs/edit?usp=sharing)
+* [Revisit container and host OS compatibility](https://docs.google.com/document/d/1lzwh8DGMu5vXXHwJmnewYIMffkcOEvH8owX4UYjRcw0/edit#heading=h.fcxt9vheg92c)
+
+## Scope
+
+* Define image compatibility spec that describes special host OS requirements of containerized application.
+* The spec should describe container requirements for Linux, FreeBSD and Windows.
+* The final list of supported fields (like kernel modules, configuration, out-of-tree drivers etc.)
+  is to be determined by the working group for each supported OS.
+* Make the change as minimally invasive as possible to the image, runtime and distribution specs - that could be achieved over
+  delivering compatibility artifact type.
+* Deprecate os.version and os.features fields from the image spec that were mainly used for Windows.
```

Comment:
This feels really out of place here - I can debate the technical merits, but I think this makes more sense as a discussion topic for the WG to propose to existing specs than as a scope item for the WG.

(Also, without folks from the relevant Microsoft team involved in the WG, I'm afraid this particular goal isn't going to progress, so we'll need to find a way to engage them in the discussions.)

Reply:
> This feels really out of place here - I can debate the technical merits, but I think this makes more sense as a discussion topic for the WG to propose to existing specs than as a scope item for the WG.

Makes sense.

> (Also, without folks from the relevant Microsoft team involved in the WG, I'm afraid this particular goal isn't going to progress, so we'll need to find a way to engage them in the discussions.)

Yes, I am aware of that. I already reached @sajayantony to ask if at Microsoft they might be interested in this. The same applies to FreeBSD community (@gtewallace). If we don't have representatives on board then support for Windows and FreeBSD, unfortunately, won't be a goal for this working group.

Reply:
> If we don't have representatives on board then support for Windows and FreeBSD, unfortunately, won't be a goal for this working group.

However, it doesn't mean they cannot join in the future after the wg is approved :)

Reply:
> This feels really out of place here - I can debate the technical merits, but I think this makes more sense as a discussion topic for the WG to propose to existing specs than as a scope item for the WG.

I think I disagree here, and think that this spec should be modular / exist on its own, allowing for a developer or user to make use of it (or not). If we throw everything into an existing spec they largely are going to get bloated. I think it's better to err on the side of having more specs that are optimized for one purpose, vs. fewer that are chonky with information that isn't always needed. Finally, what came to mind when I read this was to have this new spec be an artifact akin to an SBOM linked to an image via a reference.

Reply:
> > This feels really out of place here - I can debate the technical merits, but I think this makes more sense as a discussion topic for the WG to propose to existing specs than as a scope item for the WG.
> 
> I think I disagree here, and think that this spec should be modular / exist on its own, allowing for a developer or user to make use of it (or not). If we throw everything into an existing spec they largely are going to get bloated. I think it's better to err on the side of having more specs that are optimized for one purpose, vs. fewer that are chonky with information that isn't always needed. Finally, what came to mind when I read this was to have this new spec be an artifact akin to an SBOM linked to an image via a reference.

I agree that it should exist on its own and we should not extend the image spec with compatibility fields.
For me it would be ideal if we had a new spec delivered as an artifact, but we need to determine this within the working group based on different opinions. The deprecation of Windows fields may be a result of the wg work, but this is not certain, so removing this item from the scope is a good idea in my opinion.

Reply:
Compared to extending image-spec/distribution-spec to including image compatibility requirements, using artifact to describe image compatibility is less invassive to existing production registries and clients. Extending current image-spec/distribution spec often means have to upgrade all production registries to conform the new spec, it is not friendly.

It's up to the consensus of the proposed working group for the final spec.

There are already os.version and os.features fields in current image-spec, it's good to have windows container developers/users join the wg, to discuss how to evolve the spec.

Reply:
I'd focus on what you want to achieve, rather than how. That way the WG can discuss without be bound to this implementation.

Reply:
Right, that's why I removed the point from the scope. I consider this discussion to be resolved.
If anybody has a different opinion then please reopen.